internal/githubclt/client.go

```diff
@@ -0,0 +1,239 @@
+// Package githubclt provides a github API client.
+package githubclt
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"github.com/google/go-github/v35/github"
+	"go.uber.org/zap"
+
+	"github.com/simplesurance/goordinator/internal/goorderr"
+	"github.com/simplesurance/goordinator/internal/logfields"
+)
+
+const DefaultHTTPClientTimeout = time.Minute
+
+const loggerName = "github_client"
+
+// New returns a new github api client.
+func New(oauthAPItoken string) *Client {
+	return &Client{
+		clt:    newGHClient(oauthAPItoken),
+		logger: zap.L().Named(loggerName),
+	}
+}
+
+func newGHClient(apiToken string) *github.Client {
+	httpClt := http.Client{
+		Timeout: DefaultHTTPClientTimeout,
+	}
+
+	if apiToken == "" {
+		return github.NewClient(&httpClt)
+	}
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: apiToken},
+	)
+
+	ctx := context.WithValue(context.Background(), oauth2.HTTPClient, httpClt)
```

Comment:
That sets the http-client that is used for token acquisition.
I'm using a custom one to be able to set a timeout for the request, to prevent that it hangs forever if the server does weird things.

That it is set via the ctx seems a weird design for me, especially for an official golang pkg

Reply:
I don't know how many times I read that documentation for `oauth2.HTTPClient`, and only now I get it.

Reply:
Actually, I never acquire a token.
I'm using always the same static API token.
Therefore settings a custom http client in the context here should have no effect

Reply:
I removed that part, thanks :-)